test/e2e/service.go

```diff
@@ -1096,6 +1096,63 @@ var _ = framework.KubeDescribe("Services", func() {
 			framework.Failf("expected un-ready endpoint for Service %v within %v, stdout: %v", t.name, kubeProxyLagTimeout, stdout)
 		}
 	})
+
+	It("should only allow access from service loadbalancer source ranges [Slow]", func() {
+		// this feature currently supported only on GCE/GKE/AWS
+		framework.SkipUnlessProviderIs("gce", "gke", "aws")
+
+		loadBalancerCreateTimeout := loadBalancerCreateTimeoutDefault
+		largeClusterMinNodesNumber := 100
+		if nodes := framework.GetReadySchedulableNodesOrDie(cs); len(nodes.Items) > largeClusterMinNodesNumber {
+			loadBalancerCreateTimeout = loadBalancerCreateTimeoutLarge
+		}
+
+		namespace := f.Namespace.Name
+		serviceName := "lb-sourcerange"
+		jig := NewServiceTestJig(cs, serviceName)
+
+		By("Prepare allow source ips")
+		// prepare the exec pods
+		// hostPod and acceptPod are allowed to access the loadbalancer
+		hostPod := framework.LaunchHostExecPod(cs, namespace, "hostexec-accept")
+		acceptPodName := createExecPodOrFail(cs, namespace, "execpod-accept")
+		dropPodName := createExecPodOrFail(cs, namespace, "execpod-drop")
+
+		hostPod, err := cs.Core().Pods(namespace).Get(hostPod.Name)
```

Comment:
Why this line is needed? Is it because the `hostPod` returned above does not come with Status.HostIP?